src/main/java/fi/aalto/cs/intellij/ui/ModuleList.java

```diff
@@ -0,0 +1,72 @@
+package fi.aalto.cs.intellij.ui;
+
+import com.intellij.ui.components.JBList;
+import com.intellij.uiDesigner.core.GridConstraints;
+import com.intellij.uiDesigner.core.GridLayoutManager;
+import fi.aalto.cs.intellij.services.CourseInformation;
+
+import java.awt.Desktop;
+import java.awt.Dimension;
+import java.awt.Insets;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import javax.swing.DefaultListModel;
+import javax.swing.JPanel;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ModuleList {
+  private static final Logger logger = LoggerFactory
+      .getLogger(CourseInformation.class);
+
+  private JBList<String> modules;
+  private JPanel basePanel;
+
+  /**
+   * Initializes a panel with a list of modules.
+   */
+  public ModuleList() {
+    basePanel = new JPanel();
+    basePanel.setLayout(new GridLayoutManager(1, 1, new Insets(0, 0, 0, 0), -1, -1));
+    modules = new JBList<String>();
+
+    CourseInformation courseInformation = CourseInformation.getInstance();
+
+    List<String> moduleNames = courseInformation.getModuleNames();
```

Comment:
would it make sense to check, if the `moduleNames` are empty?

Reply:
`JBList` (the GUI component used to display the modules) displays a text with something like "Nothing to show hree" if the list is empty, which I think is good enough for now